File: app/src/main/java/com/example/foodrecipes/requests/ImageApi.java
package com.example.foodrecipes.requests;

import com.example.foodrecipes.requests.responses.CompanyLogoResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ImageApi {

    @GET("/stable/stock/{symbol}/logo")
    Call<CompanyLogoResponse> getImageUrl(
            @Path("symbol") String symbol,
            @Query("token") String token
    );
}

File: app/src/main/java/com/example/foodrecipes/requests/ImageApi.kt
package com.example.foodrecipes.requests

import com.example.foodrecipes.requests.responses.CompanyLogoResponse
import com.example.foodrecipes.requests.responses.PriceResponse
import kotlinx.coroutines.Deferred
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ImageApi {
    @GET("/stable/stock/{symbol}/logo")
    fun getImageUrl(
            @Path("symbol") symbol: String?,
            @Query("token") token: String?
    ): Deferred<CompanyLogoResponse?>?

    @GET("/stable/stock/{symbol}/book")
    fun getPrice(@Path("symbol") symbol: String?,
                 @Query("token") token: String?): Deferred<PriceResponse?>?
}